### DISCUSSION TITLE
Contribute/Setup and work Locally

### DISCUSSION THREAD

**Discussion by Kiriakos1998** [playframework/playframework#11790] [General]
I have been reading the documentation and Contribute guide but so far I have not found any detailed instructions about how to run and debug the play framework locally. Are there any resources that I missed that can help me or the only way to debug and test any changes is by building a local version of the project and then including it in a play application to actually debug the fix/feature developed?  If these resources do exist can you help me find them if not I would really appreciate it if someone can summarize some of the most important steps to work on the source code locally. So far the closest resources that I have found are this one [building from source](https://www.playframework.com/documentation/2.8.x/BuildingFromSource) and this one [CONTRIBUTING.md](https://github.com/playframework/.github/blob/main/CONTRIBUTING.md). But none of them explains anything about debugging. Also, I see the only reference to an IDE is a link to a StackOverflow issue [StackOverflow](https://stackoverflow.com/questions/10053201/how-to-setup-eclipse-ide-work-on-the-playframework-2-0/10055419#10055419) but it only explains the process about the eclipse(does this mean I can't use Intellij?).

**Comment by mkurz** [comment]
Hi!
There is a page on how to set up IDE's, also mentioning Intellij: https://www.playframework.com/documentation/2.8.x/IDE#IntelliJ-IDEA
Basically: "File" -> "New" -> "Project from existing Source", select the folder with the project (the one that contains the `build.sbt` file), then "Import project from external model" and choose sbt, on the next screen I recommend checking sbt shell for "project reload" and "builds".

To debug a Play application in IntelliJ you need to create a "Remote JVM Debug" configuration. If you e.gg set the debug port port in this dialog to 9999, you can start sbt with `sbt -jvm-debug 9999 shell` to start a debug session you can attach the ide to.

What feature do you plan to work on?

Also, actually there are open PRs to finally enhance (besides other stuff) the contribution and getting started experience. I want to unify those attempts someday:
* https://github.com/playframework/playframework/pull/11549
* https://github.com/playframework/play-meta/issues/203
* https://github.com/playframework/playframework/pull/11136
* https://github.com/playframework/playframework.com/pull/478

**Reply by Kiriakos1998** [reply]
Thank you very much for the help. I have not find yet a bug/feature to work on. I wanted to first manage build locally the source code. I will try to pick up a small bug perhaps some of the following days and solve it ( from the good-first-issue tag).
